### PR TITLE
Ignore openrc cgroup

### PR DESCRIPTION
**Pull request by derekwaynecarr**
**What this PR does / why we need it**:
It is a work-around for the following: https://github.com/opencontainers/runc/issues/1440

**Special notes for your reviewer**:
I am open to a cleaner way to do this, but we have many developer users on Macs that ran containerized kubelets that are not able to run them right now due to the inclusion of openrc tripping up our existence checks.  Ideally, runc can give us a call to say "does this exist according to what runc knows about".  Or we could add a whitelist check.  Right now, this was the smallest hack pending more discussion.

### CODE REVIEW

File: pkg/kubelet/cm/cgroup_manager_linux.go [hunk 1]
```diff
@@ -228,7 +228,12 @@ func (m *cgroupManagerImpl) Exists(name CgroupName) bool {
 	cgroupPaths := m.buildCgroupPaths(name)
 
 	// If even one cgroup path doesn't exist, then the cgroup doesn't exist.
-	for _, path := range cgroupPaths {
+	for mount, path := range cgroupPaths {
+		// See: https://github.com/opencontainers/runc/issues/1440
+		if mount == "openrc" {
+			glog.Warningf("Unsupported cgroup subsystem found: %v", mount)
```
Comment (sjenning) [inline pkg/kubelet/cm/cgroup_manager_linux.go]: I don't think we should not log this.  It has no impact on the execution path and this path is called in frequently executed paths.

Comment (derekwaynecarr) [inline pkg/kubelet/cm/cgroup_manager_linux.go]: Agree.

**Comment by vishh** [comment]
@derekwaynecarr can we avoid adding these checks in kubelet and instead just patch runc?

**Comment by derekwaynecarr** [comment]
@vishh -- i need to do something in the interim, so i am going to update this pr to a whitelist of controllers pending resolution in runc.  feel free to jump in on https://github.com/opencontainers/runc/issues/1440 with your thoughts.

**Comment by derekwaynecarr** [comment]
@vishh @sjenning -- removed logging, updated to a known whitelist pending outcome of runc discussion.

File: pkg/kubelet/cm/cgroup_manager_linux.go [hunk 2]
```diff
@@ -227,8 +227,20 @@ func (m *cgroupManagerImpl) Exists(name CgroupName) bool {
 	// Get map of all cgroup paths on the system for the particular cgroup
 	cgroupPaths := m.buildCgroupPaths(name)
 
+	// the presence of alternative control groups not known to runc confuses
+	// the kubelet existence checks.
+	// ideally, we would have a mechaninsm in runc to support Exists() logic
+	// scoped to the set control groups it understands.  this is being discussed
+	// in https://github.com/opencontainers/runc/issues/1440
+	// once resolved, we can remove this code.
+	whitelistMounts := sets.NewString("cpu", "cpuacct", "cpuset", "memory", "hugetlb", "systemd")
```
Comment (vishh) [inline pkg/kubelet/cm/cgroup_manager_linux.go]: nit: `whitelistControllers`

**Review by csrwng — APPROVED** [review]
Tested and works for me on Docker for Mac

**Comment by derekwaynecarr** [comment]
fixed the nit

**Comment by vishh** [comment]
/lgtm

Thanks @derekwaynecarr

**Pull request merged**